### PR TITLE
Logging on  aws

### PR DESCRIPTION
**Pull request by cooktheryan**
#### What does this PR do?
Adds logging components to aws ref arch

#### How should this be manually tested?
Using the CI test

#### Is there a relevant Issue open for this?
NA

#### Who would you like to review this?
cc: @e-minguez

### CODE REVIEW

**Comment by cooktheryan** [comment]
@e-minguez let me know if there are things I should add to this PR

**Comment by cooktheryan** [comment]
dang all of my elastic search containers went into crashloop

```
logging-es-cyh1q0bc-1-v602r   0/1       CrashLoopBackOff   5          6m
logging-es-p7filbgc-1-kbv30   0/1       CrashLoopBackOff   5          6m
logging-es-yplen787-1-1h2jr   0/1       CrashLoopBackOff   5          6m
```

**Comment by e-minguez** [comment]
Check this BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1466626
Basically you need to modify the configmap and add a line (it should be fixed on an errata soon)

File: reference-architecture/aws-ansible/ose-on-aws.py [hunk 1]
```diff
@@ -76,9 +76,12 @@
 @click.option('--github-client-secret', help='GitHub OAuth Client Secret')
 @click.option('--github-organization', multiple=True, help='GitHub Organization')
 @click.option('--s3-username',  help='S3 user for registry access')
-@click.option('--deploy-openshift-metrics',  help='Deploy OpenShift Metrics', type=click.Choice(['true', 'false']), default='false')
+@click.option('--openshift-hosted-metrics-deploy',  help='Deploy OpenShift Metrics', type=click.Choice(['true', 'false']), default='false')
+@click.option('--openshift-hosted-logging-deploy',  help='Deploy OpenShift Logging', type=click.Choice(['true', 'false']), default='false')
```
Comment (e-minguez) [inline reference-architecture/aws-ansible/ose-on-aws.py]: I'd say to remove the "hosted" word as it doesn't say anything besides typing more :D and make both logging and metrics enabled by default

File: reference-architecture/aws-ansible/ose-on-aws.py [hunk 2]
```diff
@@ -76,9 +76,12 @@
...
 @click.option('--openshift-hosted-metrics-storage-volume-size', default='20Gi', help='Size of OptionShift Metrics Persistent Volume',
               show_default=False)
+@click.option('--openshift-hosted-logging-storage-volume-size', default='100Gi', help='Size of OptionShift Logging Persistent Volume',
```
Comment (e-minguez) [inline reference-architecture/aws-ansible/ose-on-aws.py]: 100Gi x n-elasticsearch is a big amount of storage, but anyway, if we decide to go for 100Gi we should make this consistent in our deployments (in Azure is 10Gi)

File: reference-architecture/aws-ansible/ose-on-aws.py [hunk 3]
```diff
@@ -229,8 +234,10 @@ def launch_refarch_env(region=None,
   click.echo('\tgithub_client_id: *******')
   click.echo('\tgithub_client_secret: *******')
   click.echo('\tgithub_organization: %s' % (','.join(github_organization)))
-  click.echo('\tdeploy_openshift_metrics: %s' % deploy_openshift_metrics)
+  click.echo('\topenshift_hosted_metrics_deploy: %s' % openshift_hosted_metrics_deploy)
   click.echo('\topenshift_hosted_metrics_storage_volume_size: %s' % openshift_hosted_metrics_storage_volume_size)
+  click.echo('\topenshift_hosted_logging_deploy: %s' % openshift_hosted_logging_deploy)
```
Comment (e-minguez) [inline reference-architecture/aws-ansible/ose-on-aws.py]: Same stuff, remove the "hosted" (I don't like it :D)

File: reference-architecture/aws-ansible/playbooks/openshift-setup.yaml [hunk 4]
```diff
@@ -56,6 +57,17 @@
     openshift_hosted_registry_pullthrough: true
     openshift_hosted_registry_acceptschema2: true
     openshift_hosted_registry_enforcequota: true
+    # metrics
     openshift_hosted_metrics_storage_kind: dynamic
     openshift_metrics_hawkular_hostname: "hawkular-metrics.apps.{{ public_hosted_zone }}"
```
Comment (e-minguez) [inline reference-architecture/aws-ansible/playbooks/openshift-setup.yaml]: I think this is the default value, so maybe we should remove it

File: reference-architecture/aws-ansible/playbooks/openshift-setup.yaml [hunk 5]
```diff
@@ -45,6 +45,7 @@
       organizations: "{{ github_organization }}"
     osm_use_cockpit: true
```
Comment (e-minguez) [inline reference-architecture/aws-ansible/playbooks/openshift-setup.yaml]: Should we deploy cockpit?

**Comment by cooktheryan** [comment]
@e-minguez I'm going ti keep this in WIP until that logging bugfix happens. There really isn't any point in releasing now or shipping a work around to only remove it in the near future

File: reference-architecture/aws-ansible/playbooks/openshift-setup.yaml (hunk 5 above)
Comment (cooktheryan) [inline reference-architecture/aws-ansible/playbooks/openshift-setup.yaml]: always have by default. Some of the consultants have plugged it in and demo'd it in the field

File: reference-architecture/aws-ansible/playbooks/openshift-setup.yaml (hunk 4 above)
Comment (cooktheryan) [inline reference-architecture/aws-ansible/playbooks/openshift-setup.yaml]: dynamic is the default value or which var? somehow the review didn't post clean but i think maybe you mean the hawkular_hostname

**Comment by cooktheryan** [comment]
aos-ci-test

**Comment by cooktheryan** [comment]
aos-ci-test

**Comment by cooktheryan** [comment]
Seriously sub manager?

**Comment by cooktheryan** [comment]
aos-ci-test

**Comment by e-minguez** [comment]
aos-ci-test

**Comment by e-minguez** [comment]
Logging errata released https://bugzilla.redhat.com/show_bug.cgi?id=1466626

**Comment by e-minguez** [comment]
@cooktheryan is still WIP or it can be merged?
LGTM BTW.

File: reference-architecture/aws-ansible/playbooks/openshift-setup.yaml (hunk 5 above)
Comment (e-minguez) [inline reference-architecture/aws-ansible/playbooks/openshift-setup.yaml]: I don't like it but ok :)

**Comment by cooktheryan** [comment]
@e-minguez testing is good. If you are good with it then we can approve and ship

**Pull request merged**